microsoft-edge/progressive-web-apps-chromium/how-to/index.md

```diff
@@ -0,0 +1,319 @@
+*   Running background fetch tasks.
+*   Badging icons.
+
+Service workers are defined in a special JavaScript file, described in [Using Service Workers][MDNUsingServiceWorkers] and [Service Worker API][MDNServiceWorkerApi].
+
+To build a service worker in your project, use the **Cache-first network** service worker recipe from PWA Builder, as follows.
+
+1. Copy the source files [pwabuilder-sw-register.js][PwaBuilderServiceWorkerRegisterJS] and [pwabuilder-sw.js][PwaBuilderServiceWorkerJS] to the `public` folder in your web app project.
+
+1.  In Visual Studio Code, open `/public/index.html` and add the following code snippet inside the `<head>` tag.
+
+    ```html
+    <script type="module" src="/pwabuilder-sw-register.js"></script>
+    ```
+
+Your web app now has a service worker that uses the cache-first strategy.  The new service worker fetches resources from the cache first, and from the network only as needed.  Cached resources include images, JavaScript, CSS, and HTML.
+
+Confirm that your service worker runs, as follows:
+
+1.  Navigate to your web app using `http://localhost:3000`.  If your web app is not available, run the following command.
+
+    ```shell
+    npm start
+    ```
+
+1.  In Microsoft Edge, select `F12` to open Microsoft Edge DevTools.  Select **Application**, then **Service Workers** to view the service workers.  If the service worker is not displayed, refresh the page.
+
+    :::image type="complex" source="../media/devtools-sw-overview.png" alt-text="Microsoft Edge DevTools Service Worker overview" lightbox="../media/devtools-sw-overview.png":::
+       Microsoft Edge DevTools Service Worker overview
+    :::image-end:::
+
+1.  View the service worker cache by expanding **Cache Storage** and select **pwabuilder-precache**.  All of the resources cached by the service worker should be displayed.  The resources cached by the service worker include the app icon, app manifest, CSS, and JavaScript files.
+
+    :::image type="complex" source="../media/devtools-cache.png" alt-text="Service Worker cache in Microsoft Edge DevTools" lightbox="../media/devtools-cache.png":::
+       Service Worker cache in Microsoft Edge DevTools
+    :::image-end:::
+
+1.  Try your PWA as an offline app, as follows.  In DevTools, select **Network**, and then change the status from **Online** to **Offline**.
+
+    :::image type="complex" source="../media/devtools-offline.png" alt-text="Setting the app to offline mode in DevTools" lightbox="../media/devtools-offline.png":::
+       Setting the app to offline mode in DevTools
+    :::image-end:::
+
+1.  Refresh your app.  It should display the offline mechanism for serving the resources of your app from the cache.  
+
+    :::image type="complex" source="../media/visual-studio-nodejs-express-index.png" alt-text="PWA running offline" lightbox="../media/visual-studio-nodejs-express-index.png":::
+       PWA running offline
+    :::image-end:::
+
+<!-- todo: add an ending of the article here -->
```

Comment:
Yeah, I’d love to see a breakdown of SW registration and what the file they copied in actually does.

Reply:
Good point.
Because I didn't touch this page much during this PR, and would like to merge it sooner rather than later, I'm tempted to leave this as is for now, as well as the 2 other comments about https (the azure web app stuff) and having to install express.
Once this PR merges, I can do a follow-up one where I address these points.